resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/DriverPodKubernetesCredentialsProvider.scala

```diff
@@ -0,0 +1,100 @@
+    sparkConf: SparkConf,
+    kubernetesAppId: String) {
+
+  def getDriverPodKubernetesCredentials(): DriverPodKubernetesCredentials = {
+    val oauthTokenSecretMapping = sparkConf
+      .get(KUBERNETES_DRIVER_OAUTH_TOKEN)
+      .map(token => (DRIVER_CONTAINER_OAUTH_TOKEN_SECRET_NAME,
+         BaseEncoding.base64().encode(token.getBytes(Charsets.UTF_8))))
+    val caCertSecretMapping = convertFileConfToSecretMapping(KUBERNETES_DRIVER_CA_CERT_FILE,
+      DRIVER_CONTAINER_CA_CERT_FILE_SECRET_NAME)
+    val clientKeyFileSecretMapping = convertFileConfToSecretMapping(
+      KUBERNETES_DRIVER_CLIENT_KEY_FILE, DRIVER_CONTAINER_CLIENT_KEY_FILE_SECRET_NAME)
+    val clientCertFileSecretMapping = convertFileConfToSecretMapping(
+      KUBERNETES_DRIVER_CLIENT_CERT_FILE, DRIVER_CONTAINER_CLIENT_CERT_FILE_SECRET_NAME)
+    val secretData = (oauthTokenSecretMapping ++
+      caCertSecretMapping ++
+      clientKeyFileSecretMapping ++
+      clientCertFileSecretMapping).toMap
+    val credentialsSecret = new SecretBuilder()
+      .withNewMetadata()
+        .withName(s"$DRIVER_CONTAINER_KUBERNETES_CREDENTIALS_SECRET_NAME-$kubernetesAppId")
+        .endMetadata()
+      .withData(secretData.asJava)
+      .build()
+    val credentialsSecretVolume = new VolumeBuilder()
+      .withName(DRIVER_CONTAINER_KUBERNETES_CREDENTIALS_VOLUME_NAME)
+      .withNewSecret()
+        .withSecretName(credentialsSecret.getMetadata.getName)
+        .endSecret()
+      .build()
+    val credentialsSecretVolumeMount = new VolumeMountBuilder()
+      .withName(credentialsSecretVolume.getName)
+      .withReadOnly(true)
+      .withMountPath(DRIVER_CONTAINER_KUBERNETES_CREDENTIALS_SECRETS_BASE_DIR)
+      .build()
+    // Cannot use both service account and mounted secrets
+    sparkConf.get(KUBERNETES_SERVICE_ACCOUNT_NAME).foreach { _ =>
+      require(oauthTokenSecretMapping.isEmpty,
+        "Cannot specify both a service account and a driver pod OAuth token.")
+      require(caCertSecretMapping.isEmpty,
+        "Cannot specify both a service account and a driver pod CA cert file.")
+      require(clientKeyFileSecretMapping.isEmpty,
+        "Cannot specify both a service account and a driver pod client key file.")
+      require(clientCertFileSecretMapping.isEmpty,
+        "Cannot specify both a service account and a driver pod client cert file.")
+    }
+    DriverPodKubernetesCredentials(
+      credentialsSecret,
+      credentialsSecretVolume,
+      credentialsSecretVolumeMount)
+  }
+
+  private def convertFileConfToSecretMapping(
+      conf: OptionalConfigEntry[String],
+      secretName: String): Option[(String, String)] = {
+    sparkConf.get(conf).map(new File(_)).map { file =>
+      if (!file.isFile()) {
+        throw new SparkException(s"File provided for ${conf.key} at ${file.getAbsolutePath}" +
+          s" does not exist or is not a file.")
+      }
+      (secretName, BaseEncoding.base64().encode(Files.toByteArray(file)))
```

Comment:
The `File` object is not opened or close, but rather the stream that pulls the data has to be opened and closed, which Guava takes care of.